components/raftstore/src/store/peer.rs

```diff
@@ -1161,6 +1174,25 @@ where
         }
     }
 
+    #[inline]
+    pub fn maybe_update_apply_unpersisted_log_state(&mut self, applied_index: u64) {
+        if self.min_safe_index_for_unpersisted_apply > 0
+            && self.min_safe_index_for_unpersisted_apply < applied_index
+        {
+            self.raft_group
+                .raft
+                .set_max_apply_unpersisted_log_limit(self.max_apply_unpersisted_log_limit);
+            self.min_safe_index_for_unpersisted_apply = 0;
+        }
+    }
+
+    #[inline]
+    fn disable_apply_unpersisted_log(&mut self, min_enable_index: u64) {
```

Comment:
Can we add metrics or log so that we know when the function is not functioning, it's by design.

Reply:
done